File: backend/app/models/prescription.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text, JSON
from sqlalchemy.orm import relationship
from datetime import datetime
from app.db.database import Base

class Prescription(Base):
    __tablename__ = "prescriptions"
    
    id = Column(Integer, primary_key=True, index=True)
    appointment_id = Column(String, ForeignKey("appointments.id"), nullable=False, unique=True)
    medicines = Column(JSON, nullable=False)  # List of strings
    instructions = Column(Text, nullable=False)
    date = Column(DateTime, nullable=False, default=datetime.utcnow)
    created_at = Column(DateTime, default=datetime.utcnow)
    
    # Relationships
    appointment = relationship("Appointment", back_populates="prescription")

File: lib/models/prescription.dart
class Prescription {
  final List<String> medicines;
  final String instructions;
  final DateTime date;

  Prescription({
    required this.medicines,
    required this.instructions,
    required this.date,
  });

  Map<String, dynamic> toJson() {
    return {
      'medicines': medicines,
      'instructions': instructions,
      'date': date.toIso8601String(),
    };
  }

  factory Prescription.fromJson(Map<String, dynamic> json) {
    return Prescription(
      medicines: List<String>.from(json['medicines']),
      instructions: json['instructions'],
      date: DateTime.parse(json['date']),
    );
  }
}
